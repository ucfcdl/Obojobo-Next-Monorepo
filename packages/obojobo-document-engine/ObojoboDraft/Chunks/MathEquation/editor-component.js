import React from 'react'
import katex from 'katex'

const getLatexHtml = latex => {
	try {
		const html = katex.renderToString(latex, { displayMode: true })
		return { html }
	} catch (e) {
		return { error: e }
	}
}

class MathEquation extends React.Component {
	constructor(props) {
		super(props)
	}

	handleLatexChange(event) {
		const editor = this.props.editor
		const change = editor.value.change()
		const content = this.props.node.data.get('content')

		change.setNodeByKey(this.props.node.key, {
			data: {
				content: {
					label: content.label,
					latex: event.target.value,
					align: content.align
				}
			}
		})
		editor.onChange(change)
	}

	handleLabelChange(event) {
		const editor = this.props.editor
		const change = editor.value.change()
		const content = this.props.node.data.get('content')

		change.setNodeByKey(this.props.node.key, {
			data: {
				content: {
					label: event.target.value,
					latex: content.latex,
					align: content.align
				}
			}
		})
		editor.onChange(change)
	}

	renderLatex() {
		const content = this.props.node.data.get('content')
		let katexHtml = getLatexHtml(content.latex)
		if (katexHtml.error) {
			return (
				<div className={'katex-container katex-error'}>
					<p>{katexHtml.error.message}</p>
				</div>
			)
		}

		katexHtml = katexHtml.html

		return (
			<div className={'non-editable-chunk'}>
				<div className="katex-container" dangerouslySetInnerHTML={{ __html: katexHtml }} />
				{content.label === '' ? null : <div className="equation-label">{content.label}</div>}
			</div>
		)
	}

	renderEquationEditor() {
		const content = this.props.node.data.get('content')
		return (
			<div className={'equation-editor'}>
				<div>
					<p>Latex Equation</p>
					<input
						name={'Latex Equation'}
						value={content.latex}
						onChange={event => this.handleLatexChange(event)}
						onClick={event => event.stopPropagation()}
						frameBorder="0"
					/>
				</div>
				<div>
					<p>Equation Label</p>
					<input
						name={'Equation Label'}
						value={content.label}
						onChange={event => this.handleLabelChange(event)}
						onClick={event => event.stopPropagation()}
						frameBorder="0"
					/>
				</div>
			</div>
		)
	}

	render() {
		const content = this.props.node.data.get('content')
		return (
			<div
				className={'component obojobo-draft--chunks--math-equation pad ' + 'align-' + content.align}
			>
				{this.renderLatex()}
				{this.renderEquationEditor()}
			</div>
		)
	}
}

export default MathEquation
