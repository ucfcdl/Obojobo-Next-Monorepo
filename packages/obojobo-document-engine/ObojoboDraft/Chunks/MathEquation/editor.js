import React from 'react'
import Common from 'Common'

import emptyNode from './empty-node.json'
import Icon from './icon'
import Node from './editor-component'
import Schema from './schema'
import Converter from './converter'

const MATH_NODE = 'ObojoboDraft.Chunks.MathEquation'

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case MATH_NODE:
				return <Node {...props} {...props.attributes} />
		}
	},
	schema: Schema
}

Common.Store.registerEditorModel('ObojoboDraft.Chunks.MathEquation', {
	name: 'Math Equation',
	icon: Icon,
	isInsertable: true,
	insertJSON: emptyNode,
	slateToObo: Converter.slateToObo,
	oboToSlate: Converter.oboToSlate,
	plugins
})

const MathEquation = {
	name: MATH_NODE,
	components: {
		Node,
		Icon
	},
	helpers: {
		slateToObo: Converter.slateToObo,
		oboToSlate: Converter.oboToSlate
	},
	json: {
		emptyNode
	},
	plugins
}

export default MathEquation
