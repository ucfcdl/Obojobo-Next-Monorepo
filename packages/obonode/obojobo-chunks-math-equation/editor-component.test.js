import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { ReactEditor } from 'slate-react'
import MathEquation from './editor-component'

jest.mock('obojobo-document-engine/src/scripts/common/util/modal-util')
import { Transforms } from 'slate'
jest.mock('slate')
jest.mock('slate-react')
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper',
	() => item => item
)
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component',
	() => props => <div>{props.children}</div>
)
jest.useFakeTimers()

describe('MathEquation Editor', () => {
	beforeEach(() => {
		jest.clearAllMocks()
		jest.clearAllTimers()
	})

	test('renders with no latex', () => {
		const component = renderer.create(<MathEquation element={{ content: { latex: null } }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders with error', () => {
		global.window.katex.renderToString.mockImplementationOnce(() => {
			throw Error('Mock katex render error')
		})
		const component = renderer.create(<MathEquation element={{ content: { latex: 'x_0_0' } }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders with label', () => {
		const component = renderer.create(
			<MathEquation element={{ content: { latex: '1', label: '', size: 1 } }} />
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('opens when clicked on', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={false}
				editor={{ toggleEditable: jest.fn(), selection: 'mock-selection' }}
			/>
		)

		const container = component.find('.obojobo-draft--chunks--math-equation')

		// click to open
		container.simulate('click', { stopPropagation: jest.fn(), preventDefault: jest.fn() })

		expect(Transforms.setSelection).toHaveBeenCalled()
		expect(Transforms.select).not.toHaveBeenCalled()
	})

	test('opens and selects itself if editor has no selection', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={false}
				editor={{ toggleEditable: jest.fn(), selection: null }}
			/>
		)

		const container = component.find('.obojobo-draft--chunks--math-equation')

		// click to open
		container.simulate('click', { stopPropagation: jest.fn(), preventDefault: jest.fn() })

		expect(Transforms.setSelection).toHaveBeenCalled()
		expect(Transforms.select).toHaveBeenCalled()
	})

	test('edits properties', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)

		// click to open
		component
			.find('button')
			.at(0)
			.simulate('click')

		component
			.find('#math-equation-latex')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: 'mock math value' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		// click on size input, change, and blur
		component
			.find('#math-equation-size')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: '10' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		// click on label and change
		component
			.find('#math-equation-label')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: 'mockLabelValue' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		component
			.find('#math-equation-alt')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: 'mockAltValue' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		expect(component.state()).toMatchInlineSnapshot(`
		Object {
		  "alt": "mockAltValue",
		  "label": "mockLabelValue",
		  "latex": "mock math value",
		  "open": true,
		  "size": 10,
		}
	`)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('handles tabbing', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)
		component.find('button').simulate('keyDown', { key: 'k' })

		component.find('button').simulate('keyDown', { key: 'Tab' })

		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('edits properties', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)
		component
			.find('button')
			.at(0)
			.simulate('click')
		jest.runOnlyPendingTimers()
		component.find('button').simulate('keyDown', { key: 'k' })
		component.find('button').simulate('keyDown', { key: 'Tab' })
		jest.runOnlyPendingTimers()

		component
			.find('button')
			.at(0)
			.simulate('click')
		jest.runOnlyPendingTimers()
		component
			.find('input')
			.at(0)
			.simulate('keyDown', { key: 'k' })
		component
			.find('input')
			.at(0)
			.simulate('keyDown', { key: 'Tab', shiftKey: true })
		jest.runOnlyPendingTimers()

		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('restricts minimum size to 0.1', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)
		component
			.find('button')
			.at(0)
			.simulate('click')
		component
			.find('#math-equation-size')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: '-1' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		expect(component.state().size).toBe(0.1)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('restricts maximum size to 20', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)
		component
			.find('button')
			.at(0)
			.simulate('click')
		component
			.find('#math-equation-size')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: '22' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		expect(component.state().size).toBe(20)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('sets blank values to 1', () => {
		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={{ toggleEditable: jest.fn() }}
			/>
		)
		component
			.find('button')
			.at(0)
			.simulate('click')
		component
			.find('#math-equation-size')
			.simulate('click', { stopPropagation: jest.fn() })
			.simulate('change', { stopPropagation: jest.fn(), target: { value: '' } })
			.simulate('blur', { stopPropagation: jest.fn() })

		expect(component.state().size).toBe(1)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('freezes and unfreezes editor', () => {
		const editor = {
			toggleEditable: jest.fn()
		}

		const component = mount(
			<MathEquation
				element={{
					content: { latex: '1', label: '1.1', size: 1 }
				}}
				selected={true}
				editor={editor}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')
		component.find('#math-equation-label').simulate('focus')

		expect(editor.toggleEditable).toHaveBeenCalledWith(false)

		component.find('#math-equation-label').simulate('blur')
		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(editor.toggleEditable).toHaveBeenCalledWith(true)
	})

	test('calls setNode once edit dialog disappears', () => {
		const props = {
			editor: { toggleEditable: jest.fn() },
			element: { content: { latex: '2x/3', label: '1.1' } },
			selected: true
		}
		const component = mount(<MathEquation {...props} />)
		ReactEditor.findPath.mockReturnValueOnce('mock-path')

		expect(Transforms.setNodes).not.toHaveBeenCalled()

		component.setProps({ selected: false })

		expect(Transforms.setNodes).not.toHaveBeenCalled()
		jest.runAllTimers()

		expect(Transforms.setNodes).toHaveBeenCalledTimes(1)
		const [calledEditor, calledContent, calledAt] = Transforms.setNodes.mock.calls[0]
		expect(calledEditor).toBe(props.editor)
		expect(calledAt).toEqual({ at: 'mock-path' })
		expect(calledContent).toEqual({
			content: {
				alt: '',
				label: '1.1',
				latex: '2x/3',
				size: 1
			}
		})
	})

	test('More Info Box handles clicks', () => {
		const editor = {
			toggleEditable: jest.fn()
		}
		React.createRef = jest.fn()
		const component = mount(
			<MathEquation
				element={{ content: { latex: '2x/3', label: '1.1' } }}
				selected={true}
				editor={editor}
			/>
		)

		const nodeInstance = component.instance()
		nodeInstance.node = {
			current: {
				contains: value => value
			}
		}

		nodeInstance.handleClick({ target: true }) // click inside
		let tree = component.html()
		expect(tree).toMatchSnapshot()

		nodeInstance.node.current = { contains: value => value }
		nodeInstance.handleClick({ target: false }) // click outside
		tree = component.html()
		expect(tree).toMatchSnapshot()

		nodeInstance.node.current = null
		nodeInstance.handleClick() // click without node
		tree = component.html()
		expect(tree).toMatchSnapshot()
	})
})
