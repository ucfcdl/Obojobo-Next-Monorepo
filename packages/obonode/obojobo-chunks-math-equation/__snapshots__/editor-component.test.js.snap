// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathEquation Editor More Info Box handles clicks 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-2x/3-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-not-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Edit</button></div></div></div></div></div>"`;

exports[`MathEquation Editor More Info Box handles clicks 2`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-2x/3-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-not-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Edit</button></div></div></div></div></div>"`;

exports[`MathEquation Editor More Info Box handles clicks 3`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-2x/3-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-not-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Edit</button></div></div></div></div></div>"`;

exports[`MathEquation Editor edits properties 2`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 10em;\\">mock-katex-render-for-mock math value-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">mockLabelValue</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"mock math value\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"mockLabelValue\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"mockAltValue\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"10\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor edits properties 3`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-not-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Edit</button></div></div></div></div></div>"`;

exports[`MathEquation Editor handles tabbing 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-not-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Edit</button></div></div></div></div></div>"`;

exports[`MathEquation Editor renders with error 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="katex-container katex-error"
    >
      <p>
        Mock katex render error
      </p>
    </div>
  </div>
</div>
`;

exports[`MathEquation Editor renders with label 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="non-editable-chunk"
    >
      <div
        className="katex-container"
        dangerouslySetInnerHTML={
          Object {
            "__html": "mock-katex-render-for-1-with-options-{\\"displayMode\\":true}",
          }
        }
        style={
          Object {
            "fontSize": "1em",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`MathEquation Editor renders with no latex 1`] = `
<div>
  <div
    className="component obojobo-draft--chunks--math-equation pad align-center"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="non-editable-chunk"
    >
      <div
        className="katex-display"
      >
        <span
          className="no-equation"
        >
          No Equation (Click to edit)
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MathEquation Editor restricts maximum size to 20 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 20em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"1.1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"20\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor restricts minimum size to 0.1 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 0.1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"1.1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"0.1\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;

exports[`MathEquation Editor sets blank values to 1 1`] = `"<div><div class=\\"component obojobo-draft--chunks--math-equation pad align-center\\" contenteditable=\\"false\\"><div class=\\"non-editable-chunk\\"><div class=\\"katex-container\\" style=\\"font-size: 1em;\\">mock-katex-render-for-1-with-options-{\\"displayMode\\":true}</div><div class=\\"equation-label\\">1.1</div></div><div class=\\"attributes-box  is-open\\" contenteditable=\\"false\\"><div class=\\"box-border\\"><div class=\\"attributes-list\\"><div class=\\"attribute\\"><label for=\\"math-equation-latex\\">Latex:</label><input id=\\"math-equation-latex\\" value=\\"1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-label\\">Optional Label:</label><input id=\\"math-equation-label\\" value=\\"1.1\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-alt\\">Alt Text:</label><input id=\\"math-equation-alt\\" value=\\"\\"></div><div class=\\"attribute\\"><label for=\\"math-equation-size\\">Size:</label><input id=\\"math-equation-size\\" type=\\"number\\" step=\\"0.1\\" max=\\"20\\" min=\\"0.1\\" value=\\"1\\"></div><div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Done</button></div></div></div></div></div></div></div>"`;
